front-end/src/visualization/Visualization.jsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Flex, Button } from '@chakra-ui/react';
+import Graph from './Graph';
+import Navbar from '../Navbar';
+import { useParams } from 'react-router-dom';
+import { useState } from 'react';
+
+function Visualization() {
+    const params = useParams();
+    const username = params.username;
+    const [zoomLevel, setZoomLevel] = useState(1);
+
+    const zoomIn = () => {
+        setZoomLevel(prevZoom => Math.min(prevZoom * 1.1, 2));
+    };
+    const zoomOut = () => {
+        setZoomLevel(prevZoom => Math.max(prevZoom / 1.1, 0.5));
+    };
+    const resetZoom = () => {
+        setZoomLevel(1);
+    };
+
+    function getRandomColor() {
+        const letters = '01234567';
+        let color = '#';
+        for (let i = 0; i < 6; i++) {
+            color += letters[Math.floor(Math.random() * letters.length)];
+        }
+        return color;
+    }
```

Comment:
This is the hex representation, right? Can we consider using letters? Since hex colors can include them (e.g., #FFFFFF is white).

Reply:
> This is the hex representation, right? Can we consider using letters? Since hex colors can include them (e.g., #FFFFFF is white).

Yeah I removed the letters to make colors darker. Originally I had the letters and the high numbers, but the brighter colors made it too difficult to read labels so could be a problem when adding song titles! Something Roesha mentioned is some colors were too dark so I will now just get the mid range of letters.

Reply:
Great that makes sense.